### PR TITLE
Task 5 - done

### PR DESCRIPTION
**Pull request by terapro**

### CODE REVIEW

File: src/actions.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+export const resultsToStore = (results) => {
+  return {
+    type: C.RESULTS_TO_STORE,
+    payload: {
+      foundResults: results,
+      loadingData: false
+    }
+  }
+};
```
Comment (alex-belmach) [inline src/actions.js]: Could be simplified along with other actions  
`export const resultsToStore = results => ({  
  type: C.RESULTS_TO_STORE, 
  payload: {  
    foundResults: results,  
    loadingData: false,  
  },  
});`

File: src/components/common/button/button.js [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
-  'btn-transp-red': btnTranspRed,
-  'btn-UC': btnUC
-  });
-  return (
-    <div className={itemClass} onClick = {btnClick} id={id}>
+export class Button extends Component {
+  constructor(props) {
+    super(props);
+
+    this.callbackFunc = () => {
```
Comment (alex-belmach) [inline src/components/common/button/button.js]: Move functions out of the constructor. Also function name is not descriptive

File: src/components/common/input/__snapshots__/input.test.js.snap [hunk 3]
```diff
@@ -14,8 +14,8 @@ exports[`<Input /> Rendering should render the component 1`] = `
 exports[`<Input /> Rendering should render the component with a proper placeholder  1`] = `
 <input
   className="search-input"
-  onInput={[MockFunction]}
-  onKeyPress={[MockFunction]}
+  onInput={[Function]}
+  onKeyPress={[Function]}
```
Comment (alex-belmach) [inline src/components/common/input/__snapshots__/input.test.js.snap]: Why these functions are not mocks now?

File: src/components/common/spiner/spiner.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+export const Spiner = () => (
+  <img src={"img/spiner.gif"}  className='spiner' />
```
Comment (alex-belmach) [inline src/components/common/spiner/spiner.js]: No need for curly brackets for string

File: src/components/film/year-minutes/year-minutes.js [hunk 5]
```diff
@@ -1,19 +1,20 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-
 import {Label} from 'src/components/common/label/label.js';
 
-const YearMinutes = ({year, minutes}) => {
-  return (
+export const YearMinutes = ({year, minutes}) => (
     <div className='year-min'>
       <Label content={year.slice(0,4)} labelWhite labelBold />
```
Comment (alex-belmach) [inline src/components/film/year-minutes/year-minutes.js]: year will be `"unkn"` for default props

File: src/components/film/year-minutes/year-minutes.js [hunk 6]
```diff
@@ -1,19 +1,20 @@
...
       <Label content={minutes + ' min'} labelWhite labelBold/>
```
Comment (alex-belmach) [inline src/components/film/year-minutes/year-minutes.js]: Film with 0 min length?

File: src/actions.js (hunk 1 above)
Comment (terapro) [reply src/actions.js]: Ok, this code is refactored

File: src/components/common/button/button.js (hunk 2 above)
Comment (terapro) [reply src/components/common/button/button.js]: Thinking...

File: src/components/common/input/__snapshots__/input.test.js.snap (hunk 3 above)
Comment (terapro) [reply src/components/common/input/__snapshots__/input.test.js.snap]: the test used wrong props for the Input component. Fixed

File: src/components/common/button/button.js (hunk 2 above)
Comment (terapro) [reply src/components/common/button/button.js]: Done

File: src/components/common/spiner/spiner.js (hunk 4 above)
Comment (terapro) [reply src/components/common/spiner/spiner.js]: Done

File: src/components/film/year-minutes/year-minutes.js (hunk 5 above)
Comment (terapro) [reply src/components/film/year-minutes/year-minutes.js]: Corrected - props {year, minutes} are made required

File: src/components/film/year-minutes/year-minutes.js (hunk 6 above)
Comment (terapro) [reply src/components/film/year-minutes/year-minutes.js]: Corrected - props {year, minutes} are made required

File: src/components/film/year-minutes/year-minutes.js (hunk 5 above)
Comment (alex-belmach) [reply src/components/film/year-minutes/year-minutes.js]: And what happens if they will be missing? Can you be sure that they will be always in place?

Comment (terapro) [reply src/components/film/year-minutes/year-minutes.js]: Corrected: if they are missing the components shows nothing (as set in the defaultProps)

**Pull request merged**